src/components/BudgetExpiryAlertAndModal/data/utils.js

```diff
@@ -0,0 +1,50 @@
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import { PLAN_EXPIRY_VARIANTS, SEEN_ENTERPRISE_EXPIRATION_MODAL_COOKIE_PREFIX, SEEN_ENTERPRISE_EXPIRATION_ALERT_COOKIE_PREFIX } from './constants';
+import ExpiryThresholds from './expiryThresholds';
+
+dayjs.extend(duration);
+
+export const getExpirationMetadata = (endDateStr) => {
+  const endDate = dayjs(endDateStr);
+  const today = dayjs();
+  const durationDiff = dayjs.duration(endDate.diff(today));
+
+  const thresholdKeys = Object.keys(ExpiryThresholds).sort((a, b) => a - b);
+  const thresholdKey = thresholdKeys.find((key) => durationDiff.asDays() <= key);
+
+  if (thresholdKey === undefined) {
```

Comment:
nit: is the check against `undefined` necessary or could it be simplified to `!thresholdKey`?